src/client/common/process/environmentActivationService.ts

```diff
@@ -143,7 +143,8 @@ export class EnvironmentActivationService implements IEnvironmentActivationServi
         const title = DataScience.activatingPythonEnvironment().format(
             interpreter.displayName || getDisplayPath(interpreter.path)
         );
-        return KernelProgressReporter.wrapAndReportProgress(resource, title, () =>
+        const cancellationTokenSource = new CancellationTokenSource();
```

Comment:
This needs to be disposed

Reply:
this CancellationTokenSource isn't cancelled anywhere. Wouldn't it be better to not create it in the first place and not force the user to pass a token to wrapAndReportProgress?

Reply:
This cancellationTokenSource would be canceled if the user clicked the 'cancel' button the 'reportProgress'.